### ISSUE TITLE
Extremely slow since 1.2.0 (Atom.)

### ISSUE THREAD

**Issue by envygeeks** [atom/fuzzy-finder#153]
With the release of Atom 1.2.0 fuzzy-finder is now extremely slow.  Where before it would instantly display results and react much faster to my typing, now it delays so much that I'm done typing before it even finds the first result and I feel like I could slam my head on the table 3 or 4 times before it even discovers the result.

**Comment by envygeeks** [comment]
I've discovered it's because we had a link to a large folder with about a thousand files but I didn't notice that problem prior to 1.2.0 so I don't know if there is a perf problem somewhere.

**Comment by jeancroy** [comment]
> I didn't notice that problem prior to 1.2.0

Is 1.1.0 OK ? do you have the option for  alternate scoring enabled ?

**Comment by envygeeks** [comment]
@jeancroy yeah 1.1 was blazing fast, even with those links it was blazing fast, I didn't notice we were using those links until 1.2.0.  Alternate scoring is enabled because I default CTRL+O to fuzzy finder because it's much faster and it made things even easier to open.

**Comment by jeancroy** [comment]
Ok because I had one user [report](https://github.com/atom/fuzzaldrin/pull/22#issuecomment-157774270) that the alternate scoring was slow on his system. Can you try disabling it just to see if speed is faster ?

It would surprise me because it hasn't really changed since the one package with 1.1. And you system clearly had spec to run it.

**Comment by layonferreira** [comment]
I'm experiencing the same problem, when I turn off alternative scrolling, it gets a little faster, but still slower.

**Comment by lee-dohm** [comment]
If someone can reproduce the "slower on v1.2.0" or "slower with alternative scoring" ... can you do a CPU profile per [the instructions in the Atom Flight Manual](https://atom.io/docs/latest/hacking-atom-debugging#diagnose-performance-problems-with-the-dev-tools-cpu-profiler)? Having specific information could be quite useful.

**Comment by jeancroy** [comment]
> checking exclude VCS ignored paths option, seems to fix for me.

Two things that I can think of.
1. Ignore VCS remove large folder from your project, like node_modules.
2. There might be a bug where the indexing process spam fuzzy finder with new files found & ask to recompute matches.

**Comment by envygeeks** [comment]
I do Ruby, C and Python because I do systems, JavaScript isn't on my list of goto tools unless I'm in the front, .bundle is always in the list of files to ignore by default in all projects and we tell Atom to ignore those files too, we don't get any folders or files from `~/.bundle` or `/var/lib/gems`

**Comment by umpirsky** [comment]
Sorry for commenting on old issue, but I see it's still opened and it matches my experience.

It is slow in 1.13, and it was faster before, 1.12 iirc. Checking exclude VCS ignored paths option helps, but that's not a fix, I need to fuzzy find vendor classes because I often read/extend vendors code.

**Comment by lee-dohm** [comment]
@umpirsky If you can supply a CPU profile as I asked [above](https://github.com/atom/fuzzy-finder/issues/153#issuecomment-159971834) that would be helpful.

**Comment by umpirsky** [comment]
@lee-dohm I can try, can you double check the link in the comment, it leads to home page now?

**Comment by lee-dohm** [comment]
@umpirsky I updated the link above and you can find it here: http://flight-manual.atom.io/hacking-atom/sections/debugging/#diagnose-runtime-performance

**Comment by umpirsky** [comment]
@lee-dohm I saved, but saved file is empty for some reason, here is the screenshot:

![screenshot from 2017-02-07 18-38-14](https://cloud.githubusercontent.com/assets/208957/22703483/9cafbc4e-ed64-11e6-9290-bf6f98c99fe5.png)

**Comment by lee-dohm** [comment]
If you can try again it would be helpful. The benefit of having the CPU
profile is being able to drill down and see the numbers for ourselves and
investigate where the bottlenecks are.

**Comment by umpirsky** [comment]
@lee-dohm I tried a number of times, always empty files saved. Is there some preparation I should do in order to be able to save them?

**Comment by lee-dohm** [comment]
You're right, I'm getting the same problem (and the same SecurityError).
I'm following up with the dev team.

**Comment by umpirsky** [comment]
@lee-dohm OK, feel free to ping me for CPU profile once you figure it out, I'd be glad to provide one.

**Comment by lee-dohm** [comment]
Will do @umpirsky ... we're tracking this in https://github.com/atom/atom/issues/13767 if you would like to subscribe as well.

**Comment by umpirsky** [comment]
It got much faster after disabling `file-icons` https://github.com/atom/fuzzy-finder/issues/269.
